### ISSUE TITLE
Wildcard address for listeners

### ISSUE THREAD

**Issue by FlowShowcontrol** [adzialocha/osc-js#8]
Hi,

great plugin! I was wondering if it would be possible to have a wildcard address like this:
```
    osc.on('/*', (message) => {
        console.log(message);
    })
```

**Comment by adzialocha** [comment]
Hi @AudioW! Thank you :smiley_cat: 

Yes, wildcards are possible. The library supports OSC address pattern matching. You can read about this here: http://opensoundcontrol.org/spec-1_0-examples#addressparts or here http://opensoundcontrol.org/spec-1_0

```
-  '?' in the OSC Address Pattern matches any single character
-  '*' in the OSC Address Pattern matches any sequence of zero or more characters
- A string of characters in square brackets (e.g., "[string]") in the OSC Address Pattern matches any character in the string. Inside square brackets, the minus sign (-) and exclamation point (!) have special meanings:
- two characters separated by a minus sign indicate the range of characters between the given two in ASCII collating sequence. (A minus sign at the end of the string has no special meaning.)
- An exclamation point at the beginning of a bracketed string negates the sense of the list, meaning that the list matches any character not in the list. (An exclamation point anywhere besides the first character after the open bracket has no special meaning.)
- A comma-separated list of strings enclosed in curly braces (e.g., "{foo,bar}") in the OSC Address Pattern matches any of the strings in the list.
- Any other character in an OSC Address Pattern can match only the same character.
```

**Comment by AudioW** [comment]
ah right, that's what I meant: is there any way to respond to every osc message that comes in?

**Comment by adzialocha** [comment]
@AudioW no, thats not implemented yet since it wouldnt' follow the OSC standard (at least how I understand it), but one could think of an extension.

**Comment by breedx2** [comment]
👍 for this enhancement.

Not only would it help for troubleshooting/logging (like the requester's simple example), but also for those of us who need to do dynamic routing.

My specific use case:  I have multiple, dynamic clients connecting to my server.  Each of those clients may register an OSC namespace prefix (address prefix) for which it wants to receive messages.  Within every namespace are potentially hundreds of methods (param1, param2, ..., paramN).  So the overall address space looks like:
```
/ns1/param1
/ns1/param2
...
/ns1/paramN
/ns2/param1
/ns2/param2
...
```
If every single possible method (fully qualified address) has to have an `.on()` set up, it becomes a nightmare to manage!  I'd much rather do something like:
```
osc.on('/ns1/*', message => { 
  // look up the namespace prefix for this address
  var ns = extractNs(message);
  // dispatch message to all clients subscribed to this namespace
  clients[ns].forEach(client => client.send(message));
});
```

**Comment by breedx2** [comment]
The OSC 1.0 spec also says:
```
An OSC Server's OSC Address Space can be dynamic; that is, its contents and shape can change over time.
```
To me, anyway, that implies that servers may have a general need to pattern match.

**Comment by derwaldgeist** [comment]
+1

For me, this prevents me from using your library. I need to react on all possible OSC messages using the very same handler, since I cannot anticipate what the actual messages will be.

**Comment by adzialocha** [comment]
@derwaldgeist sure, I understand. You are happy to contribute if you feel like making a PR

**Comment by adzialocha** [comment]
The feature was shipped for version `2.0.0`
